app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,15 @@
+class AccountActivationsController < ApplicationController
+  def edit
+    user = User.find_by email: params[:email]
+
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
```

Comment:
logic ở đây hoạt động như thế nào em nhỉ ?

Reply:
Trong câu lệnh if sẽ kiểm tra đồng thời : tồn tại người dùng trong DB thông qua tìm kiếm bằng  email, kiểm tra người dùng đã được active chưa, check authenticated . !user.activated? trả về kiểu boolean, dùng để ngăn chặn truy cập trong trường hợp : Tài khoản đã được kích hoạt , nhưng người khác có được link activation , và họ sẽ có thể click vào đường link này , nếu không kiểm tra kích hoạt thì họ sẽ log in được vào tài khoản luôn , như vậy sẽ không an toàn.